src/x/sync/pooled_worker_pool_test.go

```diff
@@ -55,42 +52,52 @@ func TestPooledWorkerPoolGo(t *testing.T) {
 
 func TestPooledWorkerPoolGoWithTimeout(t *testing.T) {
 	var (
-		workers         = 2
-		channelCapacity = 1
+		workers = 2
+		opts    = NewPooledWorkerPoolOptions().SetNumShards(int64(workers))
 	)
-	// So we can control how empty the worker pool chanel is we
-	// set capacity to be same as num workers.
-	disableParallelExecutionLock.Lock()
-	pooledWorkerPoolGoroutinesCapacity = &channelCapacity
-	defer func() {
-		pooledWorkerPoolGoroutinesCapacity = nil
-		disableParallelExecutionLock.Unlock()
-	}()
-
-	p, err := NewPooledWorkerPool(workers, NewPooledWorkerPoolOptions())
+	p, err := NewPooledWorkerPool(workers, opts)
 	require.NoError(t, err)
 	p.Init()
 
-	var (
-		resultsTrue  = 0
-		resultsFalse = 0
-		wg           sync.WaitGroup
-	)
+	pooledWorkerPool, ok := p.(*pooledWorkerPool)
+	require.True(t, ok)
+
+	// First fill up all the queues without blocking.
+	var wg sync.WaitGroup
 	wg.Add(1)
-	for i := 0; i < workers*2; i++ {
+
+	// Enqueue workers * 2 since buffered channel will allow workers / shards
+	// (which is 1, since 2 / 2 = 1) which means we need to enqueue two times
+	// the workers.
+	totalEnqueue := workers * 2
+	for i := 0; i < totalEnqueue; i++ {
+		// Set now in such a way that independent shards are selected.
+		now := time.Now().
```

Comment:
nit: calling `time.Now()` in a loop may lead to slightly off results?

Reply:
True yeah, let me call that from before the loop.